cpp/src/arrow/array/data.cc

```diff
@@ -140,6 +141,49 @@ std::shared_ptr<ArrayData> ArrayData::Make(std::shared_ptr<DataType> type, int64
   return std::make_shared<ArrayData>(std::move(type), length, null_count, offset);
 }
 
+namespace {
+template <typename Fn>
+Result<std::shared_ptr<ArrayData>> copy_to_impl(const ArrayData& data,
```

Comment:
nit: `CopyToImpl`

Reply:
Can you address this admittedly minor comment @zeroshade ? :-)